rfcs/0009-abstraction-layer-for-mobile-error-tracking.md

```diff
@@ -0,0 +1,191 @@
+
+  const createTrackingFn = (fn) => (...args) => {
+    if (!isTrackingEnabled) {
+      return Promise.resolve();
+    }
+
+    return fn(...args);
+  };
+
+  return {
+    captureError: createTrackingFn(captureError),
+    addBreadcrumb: createTrackingFn(addBreadcrumb),
+    setUserId: createTrackingFn(setUserId),
+    setLevel: createTrackingFn(setLevel),
+    enable,
+  };
+};
+
+export default createErrorTracking;
+```
+
+Example for Firebase Crashlytics:
+
+```js
+import crashlytics from "@react-native-firebase/crashlytics";
+
+const captureError = crashlytics().recordError;
+
+const setUserId = crashlytics().setUserId;
+
+const setLevel = (level) => crashlytics().setAttributes({ level });
+
+// level should be: fatal, error, warning, info, or debug.
+const addBreadcrumbs = ({ scope, message, level = "info" }) =>
+  crashlytics().setAttributes({ category: scope, message, level });
+
+const enable = (enabled) => {
+  isTrackingEnabled = enabled;
+};
+
+const createErrorTracking = ({ enabled = false } = {}) => {
+  let isTrackingEnabled = enabled;
+
+  const createTrackingFn = (fn) => (...args) => {
+    if (!isTrackingEnabled) {
+      return Promise.resolve();
+    }
+
+    return fn(...args);
+  };
+
+  return {
+    captureError: createTrackingFn(captureError),
+    addBreadcrumbs: createTrackingFn(addBreadcrumbs),
+    setLevel: createTrackingFn(setLevel),
+    setUserId: createTrackingFn(setUserId),
+    enable,
+  };
+};
+
+export default createErrorTracking;
+```
+
+## Usage examples
+
+```js
+import createErrorTracking from "@moxy/react-native-sentry";
+
+const errorTracking = createErrorTracking({ isEnable: true });
+
+try {
+  aFunctionThatMightFail();
+} catch (err) {
+  errorTracking.captureError(err);
+}
+```
+
+## Regarding the initialization
+
+In my opinion the initialization should not be abstracted, since Sentry and Firebase Crashlytics have quite differente ways of being initialized.
+
+- Sentry:
+
+```js
+import * as Sentry from '@sentry/react-native';
+
+Sentry.init({
+    dsn: // Sentry's DSN,
+});
+```
+
+- Firebase Crashlytics:
+
+```js
+import crashlytics from "@react-native-firebase/crashlytics";
+
+// you can already use it inside the file
+```
+
+# Drawbacks
+
+The only drawback I can see is the initial effort to design the solution, since the maintenance should be really simple.
+
+# Alternatives
+
+The alternative would be using the API provided directly from the error tracking solutions. However, this is not a good idea as I've already commented about in the "Motivation" section.
+
+# Adoption strategy
+
+If we implement this proposal, the developers would need to check our own module which is probably going to be called `@moxy/react-native-{solution}`, where `solution` would be the name of the chosen error tracking solution (it could be `sentry` for instance) and see the available methods. I will probably add this to the RNWM documentation, to make its usage as clear as possible.
```

Comment:
The module themselves will have a README file so their API and usage will be documented. What we will do instead is to write down a recipe to walk developers through how to integrate these modules in the boilerplate.

Reply:
Ok, I see! I will rewrite this

Reply:
Just pushed a commit with the rewrite. Let me know if it's what is supposed to be this time, please.

Reply:
@acostalima could you check this, please?

Reply:
Check my latest review below 🙏